test/testParachain.t.sol

```diff
@@ -46,15 +55,204 @@ contract ParachainTest is Test {
+        StubXcmTransactorV2.TransactThroughSignedMultilocationCall memory savedData = savedDataArray[0];
+
+        assertEq(savedData.dest.parents, 1);
+        assertEq(savedData.dest.interior.length, 1);
+        assertEq(savedData.dest.interior[0], abi.encodePacked(hex"00", bytes4(fakeParaId)));
+        assertEq(savedData.feeLocation.parents, 1);
+        assertEq(savedData.feeLocation.interior.length, 1);
+        assertEq(savedData.feeLocation.interior[0], abi.encodePacked(hex"00", bytes4(fakeParaId)));
+        assertEq(savedData.transactRequiredWeightAtMost, 5000000000);
+        bytes memory call = abi.encodePacked(
+            abi.encode(fakePalletInstance),
+            hex"09",
+            fakeReporter,
+            bytes32(parachain.reverseExternal(fakeAmount)),
+            bytes20(fakeStaker)
+        );
+        assertEq(savedData.call, call);
+        assertEq(savedData.feeAmount, 10000000000);
+        assertEq(savedData.overallWeight, 9000000000);
+    }
 
-    function testReportStakeWithdrawRequested() public {}
+    function testReportStakeWithdrawRequested() public {
+        // setup
+        IRegistry.Parachain memory fakeParachain = IRegistry.Parachain({
+            id: fakeParaId,
+            owner: paraOwner,
+            palletInstance: abi.encode(fakePalletInstance),
+            stakeAmount: fakeStakeAmount
+        });
+        IRegistry.Parachain memory badFakeParachain = IRegistry.Parachain({
+            id: fakeParaId,
+            owner: address(0),
+            palletInstance: abi.encode(fakePalletInstance),
+            stakeAmount: fakeStakeAmount
+        });
+        
+        uint256 fakeAmount = 100e18;
+
+        // test non-registered parachain - should revert with "Parachain not registered"
+        vm.expectRevert();
+        vm.prank(fakeStakingContract);
+        parachain.reportStakeWithdrawRequestedExternal(badFakeParachain, fakeReporter, fakeAmount, fakeStaker);
+
+        // test registered parachain
+        vm.prank(fakeStakingContract);
+        parachain.reportStakeWithdrawRequestedExternal(fakeParachain, fakeReporter, fakeAmount, fakeStaker);
+
+        // check saved data passed to StubXcmTransactorV2 through transactThroughSigned
+        StubXcmTransactorV2.TransactThroughSignedMultilocationCall[] memory savedDataArray = xcmTransactor.getTransactThroughSignedMultilocationArray();
+        StubXcmTransactorV2.TransactThroughSignedMultilocationCall memory savedData = savedDataArray[0];
+
+        assertEq(savedData.dest.parents, 1);
+        assertEq(savedData.dest.interior.length, 1);
+        assertEq(savedData.dest.interior[0], abi.encodePacked(hex"00", bytes4(fakeParaId)));
+        assertEq(savedData.feeLocation.parents, 1);
+        assertEq(savedData.feeLocation.interior.length, 1);
+        assertEq(savedData.feeLocation.interior[0], abi.encodePacked(hex"00", bytes4(fakeParaId)));
+        assertEq(savedData.transactRequiredWeightAtMost, 5000000000);
```

Comment:
These'll have to be updated when we do benchmarking (to determine actual costs). Just fyi